### PR TITLE
Update package and fix #24

### LINKED ISSUE #24: handling of NaNs in interpolants()
What is the expected / desired behavior for the last line in this snippet? 

```
using GridInterpolations
r = GridInterpolations.RectangleGrid(1:4, 1:4)
interpolants(r, [2.5, 3.5])  # business as usual
interpolants(r, [2.5, NaN])  # expected behavior?
```
The output is `([-2, -1, 2, 3], [NaN, NaN, NaN, NaN])`.
While `NaN` weights seem to make sense, the indices seem kind of dangerous (imagine someone is just checking for the support, not the actual weights).

What would be a sane behavior? Maybe `([0], [NaN])` or `([NaN],[NaN])`? Or just checking for `NaN`s at  the beginning of the call? Throwing an error?

@MaximeBouton

### PR DESCRIPTION
**Pull request by rejuvyesh**

### CODE REVIEW

File: src/GridInterpolations.jl [hunk 1]
```diff
@@ -162,6 +162,9 @@ function interpolate(grid::AbstractGrid, data::DenseArray, x::AbstractVector)
 end
 
 function interpolants(grid::RectangleGrid, x::AbstractVector)
+    if any(isnan.(x))
+        throw(DomainError("Input contains NaN!"))
+    end
```
Comment (rejuvyesh) [inline src/GridInterpolations.jl]: This is a very simple implementation which probably allocates a bit. Another way would be iterate over `x` and bail as soon as you find a NaN. Dunno, if worth the complexity though. How large is `x` usually?

Comment (MaximeBouton) [reply src/GridInterpolations.jl]: `x` has the same size as the grid dimension so not too large. I think that is fine!

**Comment by MaximeBouton** [comment]
Thanks! @rejuvyesh

File: src/GridInterpolations.jl (hunk 1 above)
Comment (zsunberg) [reply src/GridInterpolations.jl]: @rejuvyesh `any(isnan, x)` does not have to allocate (#27). (Most reducers in julia take a function first argument, and that seems to be the most optimized form). I also tried the following (with somewhat confusing results):
- `any(isnan(t) for t in x)`: no allocations
- `any(map(isnan, x))`: allocates, but much faster than `any(isnan.(x))`
None of them allocate anything if x is a StaticVector

**Pull request merged**